misc-utils/fincore.c

```diff
@@ -124,37 +179,75 @@ static int add_output_data(struct fincore_control *ctl,
 
 	for (i = 0; i < ncolumns; i++) {
 		int rc = 0;
+		int column_id = get_column_id(i);
+		int format_value = 0;
 
-		switch(get_column_id(i)) {
+		switch(column_id) {
 		case COL_FILE:
 			rc = scols_line_set_data(ln, i, name);
 			break;
 		case COL_PAGES:
-			xasprintf(&tmp, "%jd",  (intmax_t) count_incore);
-			rc = scols_line_refer_data(ln, i, tmp);
-			break;
 		case COL_RES:
-		{
-			uintmax_t res = (uintmax_t) count_incore * ctl->pagesize;
-
-			if (ctl->bytes)
-				xasprintf(&tmp, "%ju", res);
-			else
-				tmp = size_to_human_string(SIZE_SUFFIX_1LETTER, res);
-			rc = scols_line_refer_data(ln, i, tmp);
+			value = count_incore;
+			format_value = 1;
 			break;
-		}
 		case COL_SIZE:
 			if (ctl->bytes)
 				xasprintf(&tmp, "%jd", (intmax_t) file_size);
 			else
 				tmp = size_to_human_string(SIZE_SUFFIX_1LETTER, file_size);
 			rc = scols_line_refer_data(ln, i, tmp);
+			break;
+		case COL_CACHE_PAGES:
+		case COL_CACHE:
+		case COL_DIRTY_PAGES:
+		case COL_DIRTY:
+		case COL_WRITEBACK_PAGES:
+		case COL_WRITEBACK:
+		case COL_EVICTED:
+		case COL_EVICTED_PAGES:
+		case COL_RECENTLY_EVICTED:
+		case COL_RECENTLY_EVICTED_PAGES:
+			if (!ctl->cachestat)
+				break;
+
+			switch(column_id) {
+			case COL_CACHE_PAGES:
+			case COL_CACHE:
+				value = ctl->cstat.nr_cache; break;
```

Comment:
Can we get the "break;" on a separate line? It's not readable :-)

Reply:
Maybe you can move all the second switch() to separate function, like `value = get_cstat_value(id);` to avoid switch-in-switch ;-)